apps/doji-backend/src/session/session.service.ts

```diff
@@ -0,0 +1,63 @@
+import { UserReference } from '@backend/auth/auth.service'
+import { Service } from '@backend/entities/Service'
+import { Session } from '@backend/entities/Session'
+import { User } from '@backend/entities/User'
+import {
+  GetServiceByNameAndExpertUsernameDTO,
+  ScheduleSessionDTO,
+  ServiceInformationDTO,
+} from '@backend/session/session.dto'
+import { EntityRepository, NotFoundError } from '@mikro-orm/core'
+import { InjectRepository } from '@mikro-orm/nestjs'
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common'
+
+@Injectable()
+export class SessionService {
+  constructor(
+    @InjectRepository(Session) private readonly sessionRepo: EntityRepository<Session>,
+    @InjectRepository(User) private readonly userRepo: EntityRepository<User>,
+    @InjectRepository(Service) private readonly serviceRepo: EntityRepository<Service>,
+  ) {}
+  async schedule(dto: ScheduleSessionDTO, userRef: UserReference) {
+    const service = await this.serviceRepo.findOneOrFail({
+      $and: [{ expert: { username: dto.expertUsername } }, { name: dto.serviceName }],
+    })
+    const creator = await userRef.getUser()
+    const session = new Session()
+    session.meetingProviderId = 'unknown'
+    session.fee = dto.fee
+    session.coinOnHold = dto.fee
```

Comment:
Logic Error. This does not actually withhold user's cash.

You will need to implement following in `payment/coin-transaction`

- Debit USER_PAYABLE_ACCOUNT of the host by `fee` amount (negative number)
- Credit PREPAID_EXPERT_SESSION_PAYABLE_ACCOUNT by the same amount (positive)

Once session is completed:

- Debit PREPAID_EXPERT_SESSION_PAYABLE_ACCOUNT by the fee amount (-)
- Credit USER_PAYABLE_ACCOUNT of the expert by the same amount (+)

If the user cancel the thing:

- Debit PREPAID_EXPERT_SESSION_PAYABLE_ACCOUNT by the fee amount (-)
- Credit USER_PAYABLE_ACCOUNT of the expert by the penalty fee (+)
- Credit USER_PAyABLE_ACCOUNT of the host by the rest of the fee (+)

If platform fee is needed,

- Credit PLATFORM_REVENUE_ACCOUNT by that fee.

This may be implement next spring though.